docs-sphinx/reference/smart-contract/genesis-contract.md

```diff
@@ -0,0 +1,217 @@
+# Genesis Contract
+
+## Overview
+
+Genesis Contract, also known as the Zero Contract, is mainly used to deploy and maintain smart contracts running on the aelf blockchain.
+
+This contract will be deployed first when aelf blockchain launched so that it can be used to deploy other smart contracts.
+
+To achieve this purpose, the Genesis Contract implements the following methods defined in [acs0](https://docs.aelf.io/en/latest/reference/acs/acs0.html):
+
+```protobuf
+service ACS0 {
+    // Deploy a system smart contract on chain and return the address of the system contract deployed.
+    rpc DeploySystemSmartContract (SystemContractDeploymentInput) returns (aelf.Address) {
+    }
+    
+    // Deploy a smart contract on chain and return the address of the contract deployed.
+    rpc DeploySmartContract (ContractDeploymentInput) returns (aelf.Address) {
+    }
+    
+    // Update a smart contract on chain.
+    rpc UpdateSmartContract (ContractUpdateInput) returns (aelf.Address) {
+    }
+    
+    // and others.
+    // ...
+}
+```
+
+Therefore, developers can deploy (and update) their own smart contracts by interacting with the Genesis Contract.
+
+In this article, we will discuss:
+
+- Implementation of deploy and update contracts
+- How the contract will be loaded to the smart contract execution environment
+- The current process of deploying and updating aelf smart contracts
+- Other details of the Genesis Contract
+
+## `SmartContractRegistration` and `ContractInfo`
+
+There is a critical data structure defined in `aelf/core.proto`, called SmartContractRegistration:
+
+```C#
+message SmartContractRegistration {
+    // The category of contract code(0: C#).
+    sint32 category = 1;
+    // The byte array of the contract code.
+    bytes code = 2;
+    // The hash of the contract code.
+    Hash code_hash = 3;
+    // Whether it is a system contract.
+    bool is_system_contract = 4;
+    // The version of the current contract.
+    int32 version = 5;
+    // The version of the contract.
+    string contract_version = 6;
+    // The address of the current contract.
+    Address contract_address = 7;
+    // Indicates if the contract is the user contract.
+    bool is_user_contract = 8;
+}
+```
+Smart Contract code is stored in the `code` field.
+
+However, each `SmartContractRegistration` entity is not a one-to-one correspondence with the contract, and its storage structure is:
```

Comment:
Can be rephrased, not very clear regarding the one-to-one correspondence. Only understood after reading about ContractInfo.